### PR TITLE
PCP-2952 add custom footer on error page; PCP-3323 bump up appVersion of OAuth2 Proxy; PCP-2952 change error display in DP in case of permission errors

### PR DESCRIPTION
**Pull request by snenkov-tibco**
- PCP-2952 add custom footer on error page;
- PCP-3323 bump up appVersion of OAuth2 Proxy;
- PCP-2952 change error display in DP in case of permission errors

Screen Shot 2023-12-11 at 1 33 02 PM

### CODE REVIEW

File: charts/oauth2-proxy-manifests/values.yaml [hunk 1]
```diff
@@ -96,6 +96,7 @@ extraArgs:
   skip-jwt-bearer-tokens: true
   skip-provider-button: true
   scope: openid email profile TSC
+  footer: "Secured by TIBCO Control Plane"
```
Comment (ashundi-tibco) [inline charts/oauth2-proxy-manifests/values.yaml:99]:
```suggestion
  footer: "Secured by TIBCO Platform"
```

**Review by ashundi-tibco — CHANGES_REQUESTED** [review]
1. add screenshot in description
2. merge other changes from other PR so this is clean

File: charts/oauth2-proxy-manifests/values.yaml (hunk 1 above)
Comment (snenkov-tibco) [reply charts/oauth2-proxy-manifests/values.yaml]: ok

File: charts/oauth2-proxy-manifests/templates/shell-script-cm.yaml [hunk 2]
```diff
@@ -25,22 +25,38 @@ data:
     }'
 
     exchangeIatForClientId () {
-      response=$(curl -s https://{{ .Values.global.cp.cpHostname }}/idm/v1/oauth2/clients/register --header "Authorization: Bearer {{ .Values.global.cp.secrets.iat }}" --data "${body}" --header 'Content-Type: application/json')
+      response=$(curl -f -s https://{{ .Values.global.cp.cpHostname }}/idm/v1/oauth2/clients/register --header "Authorization: Bearer {{ .Values.global.cp.secrets.iat }}" --data "${body}" --header 'Content-Type: application/json')
+      exitCode=$?
+      if [ "$exitCode" -ne 0 ]; then
+          echo "Failed to register client!"
```
Comment (ashundi-tibco) [inline charts/oauth2-proxy-manifests/templates/shell-script-cm.yaml:31]:
```suggestion
          echo "Failed to register client: ${response}"
```

File: charts/oauth2-proxy-manifests/templates/shell-script-cm.yaml [hunk 3]
```diff
@@ -25,22 +25,38 @@ data:
...
+          revokeIat
+          exit "$exitCode"
+      fi
       clientId=$(echo -n "$response" | jq -rj .client_id | base64)
       clientSecret=$(echo -n "$response" | jq -rj .client_secret | base64)
-      if [ -z "$clientId" ] || [ "$clientId" = "bnVsbA==" ] # bnVsbA== is 'null' base64encoded
-      then
-        echo "Failed to register client. Existing"
+      if [ -z "$clientId" ] || [ "$clientId" = "$(echo -n null | base64)" ]; then
+        echo "Failed to register client!"
```
Comment (ashundi-tibco) [inline charts/oauth2-proxy-manifests/templates/shell-script-cm.yaml:38]:
```suggestion
        echo "Invalid client value: ${clientId}"
```

**Review by ashundi-tibco — CHANGES_REQUESTED** [review]

**Review by ashundi-tibco — APPROVED** [review]

**Review by ashundi-tibco — APPROVED** [review]

**Pull request merged**